be/src/storage/chunk_helper.cpp

```diff
@@ -450,26 +451,34 @@ Status ChunkAccumulator::push(vectorized::ChunkPtr&& chunk) {
         }
         start += need_rows;
     }
+    _accumulate_count++;
     return Status::OK();
 }
 
 bool ChunkAccumulator::empty() const {
     return _output.empty();
 }
 
+bool ChunkAccumulator::reach_limit() const {
+    return _accumulate_count >= kAccumulateLimit;
+}
+
 vectorized::ChunkPtr ChunkAccumulator::pull() {
```

Comment:
why not use ChunkAccumulateOperator?

Reply:
This `ChunkAccumulator` could both split big chunk and merge small chunks.